### PR TITLE
Use ruff as linter

### PR DESCRIPTION
**Pull request by janbjorge**
Merge after: https://github.com/equinor/fmu-dataio/pull/404

### CODE REVIEW

File: examples/s/d/nn/_project/aggregate_surfaces.py [hunk 1]
```diff
@@ -82,8 +78,8 @@ def main():
 
         # ==============================================================================
         # Example 2: We only want the metadata (e.g. we are in a cloud service)
-        metadata = exp.generate_metadata(aggregated_surface, operation=operation)
-        print(f"Example 2: Metadata generated")
+        exp.generate_metadata(aggregated_surface, operation=operation)
```
Comment (perolavsvendsen) [inline examples/s/d/nn/_project/aggregate_surfaces.py:81]: Even if the `metadata` variable is not further used, I think it is important for the example to show that `generate_metadata()` returns metadata.

File: src/fmu/dataio/_utils.py [hunk 2]
```diff
@@ -138,7 +140,7 @@ def export_file(obj, filename, extension, flag=None):
     elif extension == ".roff" and isinstance(obj, (xtgeo.Grid, xtgeo.GridProperty)):
         obj.to_file(filename, fformat="roff")
     elif extension == ".csv" and isinstance(obj, pd.DataFrame):
-        includeindex = True if flag == "include_index" else False
+        includeindex = flag == "include_index"
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_utils.py:143]: Nice 👍

**Review by perolavsvendsen — APPROVED** [review]
👍

File: examples/s/d/nn/_project/aggregate_surfaces.py (hunk 1 above)
Comment (janbjorge) [reply examples/s/d/nn/_project/aggregate_surfaces.py:80]: Thanks, unsure why i removed this. Clearly wrong.

File: tests/conftest.py [hunk 3]
```diff
@@ -213,9 +213,9 @@ def fixture_caseglobalconfig():
 def fixture_globalconfig1():
     """Minimalistic global config variables no. 1 in ExportData class."""
 
-    cfg = dict()
+    cfg = {}
 
-    cfg = dict()
+    cfg = {}
```
Comment (mferrera) [inline tests/conftest.py:218]: Future PR could probably remove this duplication

**Pull request merged**